packages/portalnetwork/src/subprotocols/history/history.ts

```diff
@@ -1,113 +1,97 @@
-    this.logger = client.logger.extend('HistoryNetwork')
-    this.accumulator = new AccumulatorManager({ history: this })
+    this.logger = debug(this.ENR.nodeId.slice(0, 5)).extend('Portal').extend('HistoryNetwork')
     this.ETH = new ETH(this)
     this.gossipManager = new GossipManager(this)
-    this.receiptManager = new ReceiptsManager(this.client.db, this)
-    this.contentManager = new ContentManager(this, nodeRadius ?? 4n)
     this.routingTable.setLogger(this.logger)
   }
-
-  public getEpochByIndex = async (
-    index: number
-  ): Promise<Uint8Array | Uint8Array[] | undefined> => {
-    const log = this.logger.extend('portal_getEpoch')
-    log(`Searching for epoch: ${index}`)
-    const epochHash = toHexString(this.accumulator.historicalEpochs()[index])
-    log(`Searching for epoch with root: ${epochHash}`)
-    const contentKey = getHistoryNetworkContentKey(
-      HistoryNetworkContentTypes.EpochAccumulator,
-      Buffer.from(fromHexString(epochHash))
-    )
-    log(`Searching for epoch with contentKey: ${contentKey}`)
-    const contentId = getHistoryNetworkContentId(
-      HistoryNetworkContentTypes.EpochAccumulator,
-      epochHash
-    )
-    log(`Searching for epoch with contentId: ${contentId}`)
-    const lookup = new ContentLookup(this, fromHexString(contentKey))
-    const epoch = await lookup.startLookup()
-    log(`Epoch search returned: ${epoch}`)
-    return epoch
-  }
-
   /**
    *
    * @param decodedContentMessage content key to be found
    * @returns content if available locally
    */
   public findContentLocally = async (contentKey: Uint8Array) => {
-    let value = Uint8Array.from([])
-    if (contentKey[0] === HistoryNetworkContentTypes.HeaderProof) {
-      try {
-        // Create Header Proof
-        this.logger(`Creating proof for ${toHexString(contentKey.subarray(1))}`)
-        const proof = await this.generateInclusionProof(toHexString(contentKey.subarray(1)))
-        // this.logger(proof)
-        value = SszProofType.serialize({
-          leaf: proof.leaf,
-          witnesses: proof.witnesses,
-        })
-      } catch (err) {
-        this.logger(`Unable to generate Proof: ${(err as any).message}`)
-      }
-    } else {
-      try {
-        //Check to see if value in content db
-        value = Buffer.from(fromHexString(await this.client.db.get(toHexString(contentKey))))
-      } catch {}
-    }
-    return value
+    const value = await this.retrieve(toHexString(contentKey))
+    return value ? fromHexString(value) : undefined
   }
-  public init = async () => {
-    this.client.uTP.on('Stream', async (selector, blockHash, content) => {
-      if (selector === HistoryNetworkContentTypes.EpochAccumulator) {
-        blockHash = toHexString(
-          EpochAccumulator.hashTreeRoot(EpochAccumulator.deserialize(content))
-        )
-      }
-      await this.addContentToHistory(selector, blockHash, content)
+
+  public init = async () => {}
```

Comment:
I think you can safely get rid of this and then the abstract method in the base class.

Reply:
check :heavy_check_mark: